File: doc-examples/nitrite-example-java/src/main/java/example/StudentRepository.java
package example;

import io.micronaut.data.nitrite.annotation.NitriteRepository;
import io.micronaut.data.repository.CrudRepository;
import io.micronaut.data.repository.jpa.JpaSpecificationExecutor;

// tag::studentRepository[]
@NitriteRepository
public interface StudentRepository extends CrudRepository<Student, String>, JpaSpecificationExecutor<Student> {

    // tag::studentRepository-findByName[]
    Student findByName(String name);
    // end::studentRepository-findByName[]
}
// end::studentRepository[]

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/StudentRepository.kt
package example

import io.micronaut.data.nitrite.annotation.NitriteRepository
import io.micronaut.data.repository.CrudRepository
import io.micronaut.data.repository.jpa.JpaSpecificationExecutor

// tag::studentRepository[]
@NitriteRepository
interface StudentRepository : CrudRepository<Student, String>, JpaSpecificationExecutor<Student> {

    // tag::studentRepository-findByName[]
    fun findByName(name: String): Student?
    // end::studentRepository-findByName[]
}
// end::studentRepository[]
